### PR TITLE
feat(builtin): complete basic information table view for `ls`

### PR DESCRIPTION
**Pull request by Luxzi**
*See below:*

image

### CODE REVIEW

File: src/exec/builtins/functions.rs [hunk 1]
```diff
@@ -84,26 +99,163 @@ pub fn list_directory(shell: &mut ShellState, args: Vec<&str>) -> Result<()> {
...
+            } else if i.chars().last().unwrap() == '/' {
+                if i.starts_with('.') {
+                    i.as_str().dark_green()
+                } else {
+                    i.as_str().green() 
+                }
+            } else { 
+                i.as_str().white() 
+            }, 
+            format!("{}", <std::time::SystemTime as Into<DateTime<Local>>>::into(std::fs::metadata(path.to_path_buf()).unwrap().modified().unwrap()).format("%b %d %Y %T")).dark_cyan(),
```
Comment (lthoerner) [inline src/exec/builtins/functions.rs:246]: Please figure out the proper way to do this, I believe @doinkythederp mentioned it in the Discord

**Review by lthoerner — CHANGES_REQUESTED** [review]

File: src/exec/builtins/functions.rs (hunk 1 above)
Comment (doinkythederp) [reply src/exec/builtins/functions.rs:246]: Yeah you should be able to just do this:
```suggestion
            format!("{}", DateTime::<Local>::from(std::fs::metadata(path.to_path_buf()).unwrap().modified().unwrap()).format("%b %d %Y %T")).dark_cyan(),
```

Comment (Luxzi) [reply src/exec/builtins/functions.rs]: Fixed in f560871

File: src/exec/builtins/args.rs [hunk 2]
```diff
@@ -30,6 +30,12 @@ pub struct ChangeDirectoryArgs {
 pub struct ListDirectoryArgs {
     #[arg(short = 'a', long = "all", help = "Show hidden files and directories")]
     pub show_hidden: bool,
+    #[arg(short = 'l', long = "long", help = "Show files with details as a table")]
+    pub long_view: bool,
+    #[arg(short = 'o', long = "octal-permissions", help = "Use octal permissions instead of string representation")]
+    pub octal_permissions: bool,
+    #[arg(long = "permission-seperator", help = "Show sperators for each permission group (eg. Current user, current group, and other users and groups)")]
+    pub permission_seperator: bool,
```
Comment (doinkythederp) [inline src/exec/builtins/args.rs:38]: Clap can infer the long form of some of these parameters so you can probably remove the `= "..."` on `octal_permissions` and `permissions_seperator`

File: src/exec/builtins/functions.rs [hunk 3]
```diff
@@ -84,26 +100,163 @@ pub fn list_directory(shell: &mut ShellState, args: Vec<&str>) -> Result<()> {
...
+
+                let mapping = vec![
+                    format!("{0}{0}{0}", dash),
+                    format!("{0}{0}{1}", dash, x), 
+                    format!("{0}{1}{0}", dash, w), 
+                    format!("{}{}{}", dash, w, x),
+                    format!("{}{1}{1}", r, dash),
+                    format!("{}{}{}", r, dash, x),
+                    format!("{}{}{}", r, w, dash),
+                    format!("{}{}{}", r, w, x)
```
Comment (doinkythederp) [inline src/exec/builtins/functions.rs:205]: A lot of these could also be simplified to inline format:
```suggestion
                    format!("{r}{w}{x}")
```

**Review by lthoerner — APPROVED** [review]

**Review by doinkythederp — COMMENTED** [review]
Just a few minor things i noticed

File: src/exec/builtins/args.rs (hunk 2 above)
Comment (Luxzi) [reply src/exec/builtins/args.rs:38]: Fixed via b7f738c

File: src/exec/builtins/functions.rs (hunk 3 above)
Comment (Luxzi) [reply src/exec/builtins/functions.rs]: Fixed via b7f738c

File: src/exec/builtins/args.rs (hunk 2 above)
Comment (Luxzi) [reply src/exec/builtins/args.rs:38]: I think its nice for readability, but I can change it if you *really* want me to

**Review by doinkythederp — APPROVED** [review]

**Pull request merged**